jina/optimizers/flow_runner.py

```diff
@@ -22,48 +22,66 @@ def run(
         """Runs the defined flow(s).
         :param trial_parameters: parameters to be used as context
         :param workspace: directory to be used for the flows
+        :param callback: callback that will be called by the flows. Should store the evaluation results.
         """
         raise NotImplementedError
 
 
 class SingleFlowRunner(FlowRunner):
-    """Module to define and run a flow."""
+    """Module to define and run a flow.
+       `documents` maps to a parameter of the `execution_method`, depending on the method.
+       To understand to which parameter `documents` is mapped see `EXECUTION_METHOD_PARAMETER_MAPPINGS`.
+       Anyhow, the default values can be overriden by the `**kwargs` argument of the `run` method.
+    """
+
+    EXECUTION_METHOD_PARAMETER_MAPPINGS = {
```

Comment:
I do not like this much? can't we rely on simply having `input_generators`? I think is not a good thing to `depend` on sugary IO

Reply:
If we want to support the `.jsonlines` construct we would need to write a jsonline reader outside once again, even thought `Flow` already support it. I would rather leverage all the available methods on `Flow` level instead of mirroring them here once again.

Reply:
Anyhow, I agree, that this is super ugly.

Reply:
The solution now puts the user in charge. I agree with you, that the whole data parsing needs a refactoring at some point in time to put it in a dedicated place.